src/api/controllers/db-controller.ts

```diff
@@ -659,25 +660,39 @@ function parseDbTxTypeMetadata(dbTx: DbTx | DbMempoolTx): TransactionMetadata {
         dbTx.contract_call_function_name,
         () => 'Unexpected nullish contract_call_function_name'
       );
+      let functionAbi: ClarityAbiFunction | undefined;
+      const abi = dbTx.abi;
+      if (abi) {
+        const contractAbi: ClarityAbi = JSON.parse(JSON.stringify(abi));
+        functionAbi = contractAbi.functions.find(fn => fn.name === functionName);
+        if (!functionAbi) {
+          throw new Error(
+            `Could not find function name "${functionName}" in ABI for ${contractId}`
+          );
+        }
+      }
       const metadata: ContractCallTransactionMetadata = {
         tx_type: 'contract_call',
         contract_call: {
           contract_id: contractId,
           function_name: functionName,
-          function_signature: '',
+          function_signature: functionAbi ? abiFunctionToString(functionAbi) : '',
           function_args: dbTx.contract_call_function_args
-            ? readClarityValueArray(dbTx.contract_call_function_args).map(c => {
+            ? readClarityValueArray(dbTx.contract_call_function_args).map((c, fnArgIndex) => {
+                const functionArgAbi = functionAbi
+                  ? functionAbi.args[fnArgIndex++]
+                  : { name: '', type: undefined };
                 return {
                   hex: bufferToHexPrefixString(serializeCV(c)),
                   repr: cvToString(c),
-                  name: '',
+                  name: functionArgAbi.name,
                   // TODO: This stacks.js function throws when given an empty `list` clarity value.
                   //    This is only used to provide function signature type information if the contract
                   //    ABI is unavailable, which should only happen during rare re-org situations.
                   //    Typically this will be filled in with more accurate type data in a later step before
```

Comment:
Now that the contract-call parsing logic is here now, shouldn't the PR include removal of the previous functions that used to do this? So the comment 
> Typically this will be filled in with more accurate type data in a later step before being sent to client.

Would no longer be true and should be updated right?

For example, are the following functions needed anymore `parseContractsWithDbTxs(...)`, `parseContractsWithMempoolTxs(...)`, `getContractCallMetadata(...)`?

Reply:
I removed `getContractCallMetadata` as it was not being used. However, maybe due to a merge error it was added again. I'll remove that. 
As far as the `parseContractsWithDbTxs` and `parseContractsWithMempoolTxs` are concerned, this PR was made when a single transaction was retrieved from DB. Feature for multiple transactions was added later. I will update the queries for the multiple transactions feature too. Once done, we won't need the 2 functions mentioned above.